lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -1949,17 +1950,19 @@ buildSignSubmitTransaction ti db@DBLayer{..} netLayer txLayer pwd walletId
                             )
                         )
 
-            addTxSubmission walletId btx slot
+            addTxSubmission walletId builtTx slot
                 & throwWrappedErr wrapNoWalletForSubmit
 
-            postTx netLayer builtSealedTx
-                & throwWrappedErr wrapNetworkError
-                & liftIO
+            pure txWithSlot
 
-            slotToUTCTime slot
-                & interpretQuery (neverFails "slot is ahead of the node tip" ti)
-                & fmap (BuiltTx{..},)
-                & liftIO
+        postTx netLayer builtSealedTx
+            & throwWrappedErr wrapNetworkError
+            & liftIO
+
+        slotToUTCTime slot
+            & interpretQuery (neverFails "slot is ahead of the node tip" ti)
+            & fmap (BuiltTx{..},)
+            & liftIO
```

Comment:
This is now unrelated to the fix? Regardless, this seems sane to me for the sake 1) not doing potentially slow networking in `atomically` 2) still updating the `pending` set and the change state together, even though I'm not sure of the exact implications 🤔